.github/workflows/ci.yml

```diff
@@ -549,6 +549,11 @@ jobs:
         run: |
           rustup show
           cargo test --locked --release -p core-primitives --lib
+          # test pallet-teerex separately, with all feature combinations
```

Comment:
Looks like a hell to maintain :(

Reply:
https://github.com/taiki-e/cargo-hack#--feature-powerset 🤔

Reply:
Not sure if the command is ok, but I managed to get those commands running from single `cargo hack` command

```
kziemianek@kziemianek ~/p/l/litentry-parachain (p-146-parachain-pallets-tests-not-run-properly-in-ci) [1]> cargo hack test --feature-powerset --features=default --locked --release -p pallet-teerex --lib
info: running `cargo test --locked --release --lib --no-default-features --features default` on pallet-teerex (1/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks` on pallet-teerex (2/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check` on pallet-teerex (3/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check` on pallet-teerex (4/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check` on pallet-teerex (5/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check` on pallet-teerex (6/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check` on pallet-teerex (7/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check` on pallet-teerex (8/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,std` on pallet-teerex (9/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,std` on pallet-teerex (10/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,std` on pallet-teerex (11/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,std` on pallet-teerex (12/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,std` on pallet-teerex (13/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,std` on pallet-teerex (14/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,std` on pallet-teerex (15/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,std` on pallet-teerex (16/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,try-runtime` on pallet-teerex (17/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,try-runtime` on pallet-teerex (18/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,try-runtime` on pallet-teerex (19/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,try-runtime` on pallet-teerex (20/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (21/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (22/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (23/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,try-runtime` on pallet-teerex (24/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,std,try-runtime` on pallet-teerex (25/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,std,try-runtime` on pallet-teerex (26/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,std,try-runtime` on pallet-teerex (27/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,std,try-runtime` on pallet-teerex (28/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (29/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (30/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,skip-ias-check,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (31/32)
info: running `cargo test --locked --release --lib --no-default-features --features default,runtime-benchmarks,skip-ias-check,skip-scheduled-enclave-check,std,try-runtime` on pallet-teerex (32/32)
```

Reply:
It looks great - although all 32 combinations might be a bit overkill 😂 

Let me integrate it into the CI